### DISCUSSION TITLE
相同的程式碼，為什麼VS code可以成功執行，但jupyter lab不行？

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#206] [Q&A] [answered]
我在VS code與jupyter lab打同一段如下的程式碼。
# 程式碼
cost1 = 100
cost2 = 20
cost = str(cost1 + cost2)
print("楊光花了"+cost+"元買水果")
在VS code可成功執行，在jupyter lab卻不行。
# VS code
![image](https://user-images.githubusercontent.com/86505454/146679690-1e1dcd5c-7091-4b03-a5d2-714baeb532db.png)

# jupyter lab
![146679703-2f6da819-ddeb-4c10-b0a2-0fdd7b383021](https://user-images.githubusercontent.com/86505454/147410011-e7f374aa-f5ba-448a-9866-edb259651fa5.png)

怎麼會發生這種事？
❓

**Comment by Joan0518** [comment, marked as answer]
我的可以耶@@
<img width="960" alt="擷取" src="https://user-images.githubusercontent.com/90574174/146980765-71cf60fe-3d69-4197-8f8b-44663c77b43a.PNG">

**Reply by enoch-lightning** [reply]
我剛剛再試一遍，結果突然可以。好奇怪，怎麼會有時候可以，有時候不行。
![image](https://user-images.githubusercontent.com/86505454/147065649-eef68034-479b-4168-ad84-96a91b90ee1f.png)

**Comment by xCk27x** [comment]
我猜你可能是前面不小心將 str 這個名稱設定為一個字串了
![螢幕擷取畫面 (34)](https://user-images.githubusercontent.com/90547641/147406096-ce15796a-644a-44c6-a877-5e6beeea764e.png)
這樣會使 str 這個名稱指向一個字串
而 str 本來是指向一個class(應該) 所以會出錯( str 變為不可調用)

**Comment by enoch-lightning** [comment]
我剛剛發現，我jupyter lab的圖片不知道為什麼沒有顯示出來，剛剛另存影像下載，再按ctrl+c複製到上面，結果現在什麼都看不到，不知道是怎麼回事？
